interface/forms/fee_sheet/new.php

```diff
@@ -1359,49 +1254,81 @@ function setJustify(seljust) {
     $val = $plrow['title'];
     echo "    <option value='" . attr($key) . "'";
     if ($key == $pricelevel) echo ' selected';
-    echo ">" . text($val) . "</option>\n";
+    echo ">" . text(xl_list_label($val)) . "</option>\n";
   }
   echo "   </select>\n";
 }
 ?>
 
 &nbsp; &nbsp; &nbsp;
 
-<?php if (!$isBilled) { ?>
-<input type='submit' name='bn_save' value='<?php echo xla('Save');?>' />
+<?php if (!$isBilled) { // visit is not yet billed ?>
+<input type='submit' name='bn_save' value='<?php echo xla('Save');?>' 
+<?php if ($rapid_data_entry) echo " style='background-color:#cc0000';color:#ffffff'"; ?>
+/>
+<?php if ($GLOBALS['ippf_specific']) { // start ippf-only stuff ?>
+<?php if ($fs->hasCharges) { // unbilled with charges ?>
+<input type='submit' name='bn_save_close' value='<?php echo xla('Save and Checkout'); ?>' />
+<?php } else { // unbilled with no charges ?>
+<input type='submit' name='bn_save_close' value='<?php echo xla('Save and Close'); ?>' />
+<?php } // end no charges ?>
 &nbsp;
-<?php if (!$hasCharges) { ?>
-<input type='submit' name='bn_save_close' value='<?php echo xla('Mark as Billed');?>' />
+<?php } // end ippf-only ?>
+<input type='submit' name='bn_refresh' onclick='return this.clicked = true;' 
+value='<?php echo xla('Refresh');?>'>
 &nbsp;
-<?php } ?>
-<input type='submit' name='bn_refresh' value='<?php echo xla('Refresh');?>'>
+<?php } else { // visit is billed ?>
+<?php if ($fs->hasCharges) { // billed with charges ?>
+<input type='button' value='<?php echo xla('Show Receipt'); ?>'
+ onclick="top.restoreSession();location='../../patient_file/pos_checkout.php?framed=1<?php
+ echo "&ptid=" . urlencode($fs->pid) . "&enc=" . urlencode($fs->encounter); ?>'" />
```

Comment:
PHP manual says urlencode "Returns a string in which all non-alphanumeric characters except -_. have been replaced with a percent (%) sign followed by two hex digits and spaces encoded as plus (+) signs."  Consider the case of spaces in the URL which are illegal.  urlencode will deal with that whereas htmlspecialchars will not.

While is seems fine to use both, I'm having trouble seeing a threat when only the urlencoded portions can be tainted by a user.  I'm guessing the quote you reference does not assume that.  Or maybe there's a situation I'm overlooking?